src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/HaFlowStatSpec.groovy

```diff
@@ -0,0 +1,196 @@
+package org.openkilda.functionaltests.spec.stats
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.functionaltests.helpers.model.SwitchTriplet
+import org.openkilda.functionaltests.model.stats.HaFlowStats
+import org.openkilda.functionaltests.model.stats.HaFlowStatsMetric
+import org.openkilda.northbound.dto.v2.haflows.HaFlow
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPatchEndpoint
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPatchPayload
+import org.openkilda.testing.Constants
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Shared
+import spock.lang.Unroll
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue
+import static org.openkilda.functionaltests.extension.tags.Tag.LOW_PRIORITY
+import static org.openkilda.functionaltests.model.stats.Direction.FORWARD
+import static org.openkilda.functionaltests.model.stats.Direction.REVERSE
+import static org.openkilda.functionaltests.model.stats.HaFlowStatsMetric.HA_FLOW_RAW_BITS
+
+@Narrative("Verify that statistic is collected for different type of Ha-Flow")
+class HaFlowStatSpec extends HealthCheckSpecification {
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+    @Shared
+    int traffgenRunDuration = 5 //seconds
+    @Shared
+    HaFlowStats stats
+    @Shared
+    HaFlow haFlow
+    @Autowired
+    @Shared
+    HaFlowStats haFlowStats
+    @Shared
+    SwitchTriplet switchTriplet
+
+    def setupSpec() {
+        switchTriplet = topologyHelper.getSwitchTriplets(false, false).find {
+            it.ep1 != it.ep2 && it.ep1 != it.shared && it.ep2 != it.shared &&
+                    [it.shared, it.ep1, it.ep2].every { it.traffGens }
+                    && it.ep2.getTraffGens().size() > 1 // needed for update flow test
```

Comment:
Discussed offline and decided to apply the same approach as for the SwitchPairs. The appropriate task will be created for that and be a part of another PR.